underfs/local/src/main/java/alluxio/underfs/local/LocalUnderFileSystem.java

```diff
@@ -91,8 +91,10 @@ public boolean delete(String path, boolean recursive) throws IOException {
     boolean success = true;
     if (recursive && file.isDirectory()) {
       String[] files = file.list();
-      for (String child : files) {
-        success = success && delete(PathUtils.concatPath(path, child), true);
+      if (files != null) {
```

Comment:
When will `files` be null? Based on the javadoc it seems like it shouldn't be null if file.isDirectory()`